extensions/interactions/NumericExpressionInput/directives/numeric-expression-input-rules.service.ts

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Rules service for the NumericExpressionInput interaction.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+import nerdamer from 'nerdamer';
+
+import { MathInteractionsService } from 'services/math-interactions.service.ts';
+import { INumericExpressionAnswer } from 'interactions/answer-defs';
+import { INumericExpressionRuleInputs } from 'interactions/rule-input-defs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NumericExpressionInputRulesService {
+  MatchesExactlyWith(
+      answer: INumericExpressionAnswer,
+      inputs: INumericExpressionRuleInputs): boolean {
+    let mis = new MathInteractionsService();
+
+    // Inserting '*' signs wherever necessary, if not present.
+    answer = mis.insertMultiplicationSigns(answer);
```

Comment:
Instead of removing multiplication signs in the service, can you do that in the directive for editor and viewer itself? Since that's where `replaceAbsSymbolWithText` is also done, right? Basically answers shouldn't be manipulated in the service, everything should be more or less valid (except for the tests here) before coming in, otherwise  whenever you add a new function here, this has  to be added there as well.

Reply:
I understand, but the thing is, unlike the `replaceAbsSymbolWithText` method, the `insertMultiplicationSigns` method requires the input to the method to represent a valid expression since it passes in the input to nerdamer, so if it's invalid, nerdamer will throw an error.
Also, we're not exactly manipulating the input in the rules service, we are just temporarily re-formatting the given input so that it conforms to our validations. For eg. `5sqrt(3)` is not syntactically invalid, it is just not in the format that is compatible with the validators we've written, which is why we need to temporarily reformat it.
Does that seem reasonable enough, @aks681 ?

Reply:
@aks681 
Update: I've made the changes you asked for as per discussion.